integration/tests/cook/test_basic.py

```diff
@@ -206,26 +223,23 @@ def test_progress_update_submit(self):
                                          executor=job_executor_type, max_runtime=60000)
         self.assertEqual(201, resp.status_code, msg=resp.content)
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
-        self.assertEqual(1, len(job['instances']))
-        message = json.dumps(job['instances'][0], sort_keys=True)
-        self.assertEqual('success', job['instances'][0]['status'], message)
+        message = json.dumps(job['instances'], sort_keys=True)
+        self.assertIn('success', (i['status'] for i in job['instances']), message)
 
-        job = util.wait_for_sandbox_directory(self.cook_url, job_uuid)
-        message = json.dumps(job['instances'][0], sort_keys=True)
-        self.assertIsNotNone(job['instances'][0]['output_url'], message)
-        self.assertIsNotNone(job['instances'][0]['sandbox_directory'], message)
+        instance = util.wait_for_sandbox_directory(self.cook_url, job_uuid)
+        message = json.dumps(instance, sort_keys=True)
+        self.assertIsNotNone(instance['output_url'], message)
+        self.assertIsNotNone(instance['sandbox_directory'], message)
 
-        if job_executor_type == 'cook':
+        if instance['executor'] == 'cook':
             util.sleep_for_publish_interval(self.cook_url)
-
-            job = util.wait_for_exit_code(self.cook_url, job_uuid)
-            message = json.dumps(job['instances'][0], sort_keys=True)
-            self.assertEqual(0, job['instances'][0]['exit_code'], message)
-
-            job = util.load_job(self.cook_url, job_uuid)
-            message = json.dumps(job['instances'][0], sort_keys=True)
-            self.assertEqual(25, job['instances'][0]['progress'], message)
-            self.assertEqual('Twenty-five percent in progress.txt', job['instances'][0]['progress_message'], message)
+            instance = util.wait_for_exit_code(self.cook_url, job_uuid)
+            message = json.dumps(instance, sort_keys=True)
+            self.assertEqual(0, instance['exit_code'], message)
+            self.assertEqual(25, instance['progress'], message)
+            self.assertEqual('Twenty-five percent in progress.txt', instance['progress_message'], message)
+        else:
+            self.logger.info(f'Bailing out because the cook executor is not being used: {instance}')
 
     def test_configurable_progress_update_submit(self):
```

Comment:
I think we should skip this test if cook executor is not in use.

Reply:
Done